internal/services/applications/application_resource.go

```diff
@@ -248,6 +342,25 @@ func applicationResource() *schema.Resource {
 				},
 			},
 
+			"public_client": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"redirect_uris": {
```

Comment:
makes sense to pull this out to a top level property `public_client_redirect_uris`